docs/website/docs/plus/ecosystem/iceberg.md

```diff
@@ -25,33 +87,112 @@ def my_resource():
 pipeline = dlt.pipeline("loads_iceberg", destination="iceberg")
 
 ```
+</TabItem>
+</Tabs>
 
-## Table format
-`iceberg` destination automatically assigns `iceberg` table format to all resources that it will load. You can still
-fall back to storing files (as specified in `file_format`) by setting `table_format` to **native** on a resource.
+Or on the `pipeline.run` level: <!-- can this also be defined in the yaml??-->
 
-## Configuration
-Iceberg destinations looks for its configuration under **destination.iceberg**. Otherwise it is configured
-in the same way as `filesystem` destination.
+```py
+pipeline.run(write_disposition={"disposition": "merge", "strategy": "delete-insert"})
+```
 
-```toml
-[destination.iceberg]
-bucket_url = "s3://[your_bucket_name]" # replace with your bucket name,
+## Partitioning
 
-[destination.iceberg.credentials]
-aws_access_key_id = "please set me up!" # copy the access key here
-aws_secret_access_key = "please set me up!" # copy the secret access key here
-```
+Iceberg tables can be partitioned (using [hidden partitioning](https://iceberg.apache.org/docs/latest/partitioning/)) by specifying one or more partition column hints on the source/resource level:
+
+<Tabs values={[{"label": "dlt.yml", "value": "yaml"}, {"label": "Python", "value": "python"}]}  groupId="language" defaultValue="yaml">
+  <TabItem value="yaml">
+
+  ```yaml
+  sources:
+    my_source:
+      type: sources.my_source
+      with_args:
+        columns:
+          foo:
+            partition: True
+  ```
+
+  </TabItem>
+  <TabItem value="python">
+
+  ```py
+  @dlt.resource(
+    columns={"foo": {"partition": True}}
+  )
+  def my_resource():
+      ...
+
+  pipeline = dlt.pipeline("loads_iceberg", destination="iceberg")
+  ```
+
+  </TabItem>
+</Tabs>
+
+:::caution
+Partition evolution (changing partition columns after a table has been created) is not supported.
+:::
+
+## Catalogs
+
+dlt+ uses single-table, ephemeral, in-memory, sqlite-based Iceberg catalogs. These catalogs are created "on demand" when a pipeline is run, and do not persist afterwards. If a table already exists in the filesystem, it gets registered into the catalog using its latest metadata file. This allows for a serverless setup. It is currently not possible to connect your own Iceberg catalog, but this is a feature in development.
```

Comment:
Let's tease, that milti-vendor catalogs support is coming (Polaris/Unity Catalog)